fastlane/lib/fastlane/fast_file.rb

```diff
@@ -313,8 +314,9 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
               # so we can switch between tags or branches instantly, or else,
               # it would defeat the caching's purpose.
               depth = is_eligible_for_caching ? "" : "--depth 1"
-
-              Actions.sh("git clone #{url.shellescape} #{clone_folder.shellescape} #{depth} --no-checkout #{branch_option}")
+              command = "git clone #{url.shellescape} #{clone_folder.shellescape} #{depth} --no-checkout #{branch_option}"
+              command << " -c http.extraheader='Authorization: Basic #{git_basic_authorization}'" unless git_basic_authorization.nil?
```

Comment:
Since `Basic` is only one of [several authorization schemes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes), and the `Authorization` itself is only one of [many HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers), I would be inclined to allow users to specify any header for this action.

Maybe others need a different kind of authorization, or custom [caching](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#caching), etc.

As such, please rename `git_basic_authorization` to something like `git_extra_header`, and update the docs accordingly (for example, `git_basic_authorization: "Authorization: Basic <YOUR BASE64 KEY>"` instead of `git_basic_authorization: "<YOUR BASE64 KEY>"`).